packages/@react-stately/calendar/src/useRangeCalendarState.ts

```diff
@@ -92,10 +92,12 @@ export function useRangeCalendarState<T extends DateValue = DateValue>(props: Ra
 
   // If the visible range changes, we need to update the available range.
   let lastVisibleRange = useRef(calendar.visibleRange);
-  if (!isEqualDay(calendar.visibleRange.start, lastVisibleRange.current.start) || !isEqualDay(calendar.visibleRange.end, lastVisibleRange.current.end)) {
-    updateAvailableRange(anchorDate);
-    lastVisibleRange.current = calendar.visibleRange;
-  }
+  useEffect(() => {
+    if (!isEqualDay(calendar.visibleRange.start, lastVisibleRange.current.start) || !isEqualDay(calendar.visibleRange.end, lastVisibleRange.current.end)) {
+      updateAvailableRange(anchorDate);
+      lastVisibleRange.current = calendar.visibleRange;
+    }
+  });
```

Comment:
Would be nice to avoid an extra render here too. This one seems like `lastVisibleRange` could be a `useState` also so we can update it in render and it would cause the same number of renders as before.